doc/source/data/key-concepts.rst

```diff
@@ -1,107 +1,149 @@
 .. _data_key_concepts:
 
-============
 Key Concepts
 ============
 
-.. _dataset_concept:
+Datasets
+--------
 
-----------
-Dataset
-----------
+Ray Data's main abstraction is a :class:`Dataset <ray.data.Dataset>`, which
+is a distributed data transformation pipeline. Ray Data provides APIs for loading
+data and streaming transformations in parallel.
```

Comment:
I think we want to reword this. Something like below, with the appropriate links

Ray Data's main abstraction is a Dataset which is a distributed collection of data.

Ray Data provides APIs for loading large amounts of data, transforming data in parallel, and consuming data.